tests/integration/targets/cloud_discovery_providers_azure_info/tasks/main.yml

```
@@ -0,0 +1,88 @@
+          - azure_provider_info.objects[0].credential_preference.credential_type == "dynamic"
+          - azure_provider_info.objects[0].tags.location == "site-1"
+
+    - name: Get Azure cloud discovery provider by ID
+      infoblox.universal_ddi.cloud_discovery_providers_info:
+          id: "{{ azure_provider.id }}"
+      register: azure_provider_info_by_id
+    - assert:
+        that:
+          - azure_provider_info_by_id.objects | length == 1
+          - azure_provider_info_by_id.objects[0].id == azure_provider.id
+
+    - name: Get Azure cloud discovery provider by filter query
+      infoblox.universal_ddi.cloud_discovery_providers_info:
+        filter_query: "name=='{{ cloud_discovery_provider_azure_name }}'"
+      register: azure_provider_info_by_query
+    - assert:
+        that:
+          - azure_provider_info_by_query.objects | length == 1
+          - azure_provider_info_by_query.objects[0].name == cloud_discovery_provider_azure_name
+
+    - name: Get Azure cloud discovery provider by tag filters
+      infoblox.universal_ddi.cloud_discovery_providers_info:
+        tag_filters:
+          location: "site-1"
+      register: azure_provider_info_by_tag
+    - assert:
+        that:
+          - azure_provider_info_by_tag.objects | length == 1
+          - azure_provider_info_by_tag.objects[0].name == cloud_discovery_provider_azure_name
+
+  always:
+     - name: "Delete an Azure cloud discovery provider"
+       infoblox.universal_ddi.cloud_discovery_providers:
+         name: "{{ cloud_discovery_provider_azure_name }}"
+         provider_type: "Microsoft Azure"
+         account_preference: "single"
+         credential_preference:
+           access_identifier_type: "tenant_id"
+           credential_type: "dynamic"
+         source_configs:
+           - credential_config:
+               access_identifier: "{{ config_access_id }}"
+         state: absent
+         ignore_errors: true
```

Ignore errors has incorrect indentation , it should come in line with provider name (infoblox.universal.....)

---

done